### PR TITLE
Use `get_the_post_thumbnail` function in the cover block. 

### PR DESCRIPTION
**Pull request by spacedmonkey**
## What?
Follow on from https://github.com/WordPress/gutenberg/pull/39658. 

## Why?
Improved logic, to that if cover is not an image, then do not both to load feature image. 
Ensure that `get_the_post_thumbnail` function is called, so that filters are run correctly. 
Ensure that `update_post_thumbnail_cache` is called, so that feature image are loaded into memory.

### CODE REVIEW

File: packages/block-library/src/cover/index.php
Comment (felixarntz) [inline packages/block-library/src/cover/index.php:57]: This was in fact incorrect and could result in accessibility problems, since it was using the caption as the `alt` attribute, when in fact WordPress images have their own dedicated field to define the alternative text.

So fixing that is another benefit of this PR.

File: packages/block-library/src/cover/index.php
Comment (felixarntz) [inline packages/block-library/src/cover/index.php:33]: This makes total sense, we shouldn't manually create `img` tags in WordPress core when dealing with attachments.

**Review by felixarntz — COMMENTED** [review]
@spacedmonkey Great catch! This does not only come with improvements for consistency and performance, but also accessibility. 🎉 

Please check the CI failures. Other than that I'm happy to approve it.

**Comment by adamsilverstein** [comment]
Excellent, thanks for working on this @spacedmonkey!

File: packages/block-library/src/cover/index.php
Comment (spacedmonkey) [inline packages/block-library/src/cover/index.php:44]: Copied from.
https://github.com/WordPress/wordpress-develop/blob/62040b8b9d03b7dec7fbd66c2b415504b0f5ef47/src/wp-includes/post-thumbnail-template.php#L185-L187

**Review by peterwilsoncc — APPROVED** [review]
This looks good to me. 

The change adds several attributes to the image tag
* `loading`
* `srcset`
* `sizes`
* `width`
* `height`
*  `class` changes from `wp-block-cover__image-background` to `wp-block-cover__image-background wp-post-image`

I think all these are good things provided the width and height don't affect object fit.

No changes when the image is used as a background URL.

**Comment by aristath** [comment]
I fixed a couple of PHP Coding-Standards issues, once test pass this can be merged 👍

**Review by aristath — APPROVED** [review]

File: packages/block-library/src/cover/index.php
Comment (draganescu) [reply packages/block-library/src/cover/index.php:57]: Great catch, I think it resulted like this as I didn't consider that featured images are their own kind of entity in WP, instead treated it as another simple image added to the cover block.

**Review by draganescu — APPROVED** [review]
I just want to leave the 3rd approval here :D

**Comment by gziolo** [comment]
I cherry-picked this PR for WordPress 6.0 RC2 with [93befd1](https://github.com/WordPress/gutenberg/commit/93befd1be919e55a2d789360258a31abf0650a91).

**Pull request merged**
